compiler/sem/semstmts.nim

```diff
@@ -324,6 +324,23 @@ proc identWithin(n: PNode, s: PIdent): bool =
     if identWithin(n[i], s): return true
   result = n.kind == nkSym and n.sym.name.id == s.id
 
+proc getIdentLineInfo(n: PNode): TLineInfo =
+  ## Returns the line information of the identifier-like node in the
+  ## (semantically valid) AST `n` appearing in a name slot.
+  var n {.cursor.} = n
+  while true:
```

Comment:
```suggestion
  while n.kind notin {nkIdent, nkSym}:
```

Don't have to, I just get a bit nervous about infinite loops, at least this way it'll likely npe/assert pretty quick.

Reply:
I manually unrolled the loop. The `if` chain encodes the expectations much better than the previous `while` loop.